cl/api/models.py

```diff
@@ -1,10 +1,64 @@
+from enum import Enum
+
 from django.conf import settings
+from django.contrib.auth.models import User
+from django.db import models
 from django.db.models.signals import post_save
 from django.dispatch import receiver
 from rest_framework.authtoken.models import Token
 
+from cl.lib.models import AbstractDateTimeModel
+
 
 @receiver(post_save, sender=settings.AUTH_USER_MODEL)
 def create_auth_token(sender, instance=None, created=False, **kwargs):
     if created:
         Token.objects.create(user=instance)
+
+
+class WebhookEventType(Enum):
+    RECAP_EMAIL = "Recap Email"
+
+    @classmethod
+    def choices(cls):
+        return [
+            (
+                key.value,
+                key.value,
+            )
+            for key in cls
+        ]
+
+
+class Webhook(AbstractDateTimeModel):
+    user = models.ForeignKey(
+        User,
+        help_text="The user that has provisioned the webhook.",
+        related_name="webhooks",
+        on_delete=models.CASCADE,
+    )
+    event_type = models.CharField(
+        max_length=20,
+        choices=WebhookEventType.choices(),
+    )
+    url = models.URLField(max_length=2000)
+    enabled = models.BooleanField(default=False)
+    # TODO - How do we expect to version the Webhooks?
+    webhook_version = models.CharField(max_length=10)
+    failure_count = models.IntegerField(default=0)
+
+
+class WebhookEvent(AbstractDateTimeModel):
+    webhook = models.ForeignKey(
+        Webhook,
+        help_text="The Webhook this event is associated with.",
+        related_name="webhook_events",
+        on_delete=models.CASCADE,
+    )
+    status_code = models.IntegerField()
+    content = models.CharField(max_length=5000)
+    response = models.CharField(max_length=5000)
```

Comment:
It's probably easier to use a text field here. That way if we find that 5k chars isn't enough we can change it without a DB migration. Postgres uses the same data structures for varchar and text anyway so this doesn't impact performance.